starport/services/network/networkchain/prepare.go

```diff
@@ -0,0 +1,211 @@
+package networkchain
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+	"github.com/tendermint/starport/starport/pkg/cosmosutil"
+	"github.com/tendermint/starport/starport/pkg/events"
+	"github.com/tendermint/starport/starport/services/network/networktypes"
+)
+
+// Prepare prepares the chain to be launched from genesis information
+func (c Chain) Prepare(ctx context.Context, gi networktypes.GenesisInformation) error {
+	// chain initialization
+	chainHome, err := c.chain.Home()
+	if err != nil {
+		return err
+	}
+
+	_, err = os.Stat(chainHome)
+
+	// nolint:gocritic
+	if os.IsNotExist(err) {
+		// if no config exists, perform a full initialization of the chain with a new validator key
+		if err := c.Init(ctx); err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	} else {
+		// if config and validator key already exists, build the chain and initialize the genesis
+		c.ev.Send(events.New(events.StatusOngoing, "Building the blockchain"))
+		if _, err := c.chain.Build(ctx, ""); err != nil {
+			return err
+		}
+		c.ev.Send(events.New(events.StatusDone, "Blockchain built"))
+
+		c.ev.Send(events.New(events.StatusOngoing, "Initializing the genesis"))
+		if err := c.initGenesis(ctx); err != nil {
+			return err
+		}
+		c.ev.Send(events.New(events.StatusDone, "Genesis initialized"))
+	}
```

Comment:
```suggestion
	switch {
	case os.IsNotExist(err):
		// if no config exists, perform a full initialization of the chain with a new validator key
		if err := c.Init(ctx); err != nil {
			return err
		}
	case err != nil:
		return err
	default:
		// if config and validator key already exists, build the chain and initialize the genesis
		c.ev.Send(events.New(events.StatusOngoing, "Building the blockchain"))
		if _, err := c.chain.Build(ctx, ""); err != nil {
			return err
		}
		c.ev.Send(events.New(events.StatusDone, "Blockchain built"))

		c.ev.Send(events.New(events.StatusOngoing, "Initializing the genesis"))
		if err := c.initGenesis(ctx); err != nil {
			return err
		}
		c.ev.Send(events.New(events.StatusDone, "Genesis initialized"))
	}
```

We can avoid the golint flag using the switch statement, which is clearer. it's up to you, not a blocker

Reply:
I added the `if else` voluntarily because it is easier to read to me but I can change it if you think it's the opposite

Reply:
In my IMHO the switch statement is more clear. What do you think, @ilgooz?

Reply:
switch-case I think, but can be addresses later during another refactoring

Reply:
since we have other change requests, maybe we can also include this to this pr.